test/functional/feature_reduced_data_utxo_height.py

```diff
@@ -0,0 +1,466 @@
+#!/usr/bin/env python3
+# Copyright (c) 2025 The Bitcoin Knots developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test REDUCED_DATA soft fork UTXO height checking.
+
+This test verifies that the REDUCED_DATA deployment correctly exempts UTXOs
+created before ReducedDataHeightBegin from reduced_data script validation rules,
+as implemented in validation.cpp.
+
+Test scenarios:
+1. Old UTXO (created before activation) spent during active period with violation - should be ACCEPTED (EXEMPT)
+2. New UTXO (created during active period) spent with violation - should be REJECTED
+3. Mixed inputs (old + new UTXOs) in same transaction
+4. Boundary test: UTXO created at exactly ReducedDataHeightBegin
+"""
+
+from io import BytesIO
+
+from test_framework.blocktools import (
+    COINBASE_MATURITY,
+    create_block,
+    create_coinbase,
+    add_witness_commitment,
+)
+from test_framework.messages import (
+    COIN,
+    COutPoint,
+    CTransaction,
+    CTxIn,
+    CTxInWitness,
+    CTxOut,
+)
+from test_framework.p2p import P2PDataStore
+from test_framework.script import (
+    CScript,
+    OP_TRUE,
+    OP_DROP,
+    hash256,
+)
+from test_framework.script_util import (
+    script_to_p2wsh_script,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+)
+from test_framework.wallet import MiniWallet
+
+
+# BIP9 constants for regtest
+BIP9_PERIOD = 144  # blocks per period in regtest
+BIP9_THRESHOLD = 108  # 75% of 144
```

Comment:
Suggest using a threshold of only 55% for this. Miner signalling is for readiness, not voting. If the community supports this (required for _any_ deployment), the only thing it should delay for is miner readiness, which 55% is sufficient for. Once the threshold is reached, the remaining miners still have 2 weeks of LOCKED_IN to upgrade.